### PR TITLE
Marathon.Adapter: turn apps + tasks into Envoy protobufs

### PR DESCRIPTION
**Pull request by JayH5**
- [x] CDS: App + port index
- [x] EDS: App + tasks + port index
- [x] RDS: Listener name + app + port index

### CODE REVIEW

File: test/relay/marathon/adapter_test.exs [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+                                address: "10.70.4.100",
+                                port_specifier: {:port_value, 15979}
+                              }}
+                         }
+                       }
+                     }
+                   ]
+                 }
+               ]
+             } = cla
```
Comment (JayH5) [inline test/relay/marathon/adapter_test.exs]: This is after running `mix format` 😕. I could _not_ do that...

File: lib/relay/marathon/adapter.ex [hunk 2]
```diff
@@ -68,18 +69,22 @@ defmodule Relay.Marathon.Adapter do
...
   @doc """
   Create a ClusterLoadAssignment for the given app, tasks, and port index. The
-  ClusterLoadAssignment will have the minimum amount of options set. Additional
-  options can be specified using `options`.
+  ClusterLoadAssignment will have the minimum amount of options set.
+
+  Additional options can be specified using `options` and options for nested
+  types are nested within that:
+  - ClusterLoadAssignment: `options`
+  - LocalityLbEndpoints: `options.llbe_options`
```
Comment (jerith) [inline lib/relay/marathon/adapter.ex]: Do we want to keep the full name for the option key and only abbreviate the local variable? We use full names everywhere else, but this one's a bit obnoxiously long. :-/

File: lib/relay/marathon/adapter.ex [hunk 3]
```diff
@@ -68,18 +69,22 @@ defmodule Relay.Marathon.Adapter do
...
-    {locality_lb_endpoints_options, options} =
-      Keyword.pop(options, :locality_lb_endpoints_options, [])
+    {llbe_options, options} = Keyword.pop(options, :llbe_options, [])
 
     ClusterLoadAssignment.new(
       [
         cluster_name: "#{app_id}_#{port_index}",
         endpoints:
-          task_port_locality_lb_endpoints(tasks, port_index, locality_lb_endpoints_options)
+          task_port_locality_lb_endpoints(tasks, port_index, llbe_options)
```
Comment (jerith) [inline lib/relay/marathon/adapter.ex]: Can this go on the previous line with `endpoints:` now?

File: lib/relay/marathon/adapter.ex (hunk 2 above)
Comment (jerith) [reply lib/relay/marathon/adapter.ex]: Maybe we should replace the `_options` suffix with `_opts` everywhere?

**Comment by JayH5** [comment]
This has been merged... GitHub is being weird 😕

**Pull request closed without merging**